src/gate/parserGnet/parserGnet.cpp

```diff
@@ -0,0 +1,366 @@
+        GateId data = inputs.find(lowerField)->second;
+        GateId clock = inputs.find(upperField)->second;
+        if (typeRTLIL.find(upperField)->second == "_DFF_P_") {
+          return net.addDff(data, clock);
+        } else {
+          inputs_.push_back(Gate::Signal::always(data));
+          inputs_.push_back(Gate::Signal::negedge(clock));
+          return net.addGate(GateSymbol::DFF, inputs_);
+        }
+      }
+      if (curr == GateSymbol::DFFrs) {
+        std::string type = typeRTLIL.find(upperField)->second;
+        inputs_ = typeDffsr(type, upperField, cell, inputs);
+        return net.addGate(GateSymbol::DFFrs, inputs_);
+      }
+    }
+  }
+  bool flag1 = 0, flag2 = 0;
+  size_t leftLeaf = cell.find(root)->second.first;
+  size_t rightLeaf = cell.find(root)->second.second;
+  Gate::SignalList inputs_;
+  // If leaf is a wire of pin type, then flag == 1, else it is just wire.
+  if (inputs.find(leftLeaf) != inputs.end()) {
+    flag1 = 1;
+  }
+  if (inputs.find(rightLeaf) != inputs.end()) {
+    flag2 = 1;
+  }
+  GateSymbol func = typeFunc.find(root)->second;
+  if (flag1 && flag2) {
+    // The condition is needed to check neg operator for the cell.
+    // leftLeaf == rightLeaf, when the cell contains of neg operator.
+    if (leftLeaf != rightLeaf) {
+      GateId leftLeaf_ = inputs.find(leftLeaf)->second;
+      inputs_.push_back(Gate::Signal::always(leftLeaf_));
+      GateId rightLeaf_ = inputs.find(rightLeaf)->second;
+      inputs_.push_back(Gate::Signal::always(rightLeaf_));
+    } else {
+      GateId oneLeaf = inputs.find(leftLeaf)->second;
+      inputs_.push_back(Gate::Signal::always(oneLeaf));
+    }
+    return net.addGate(func, inputs_);
+  }
+  if (flag1 && !flag2) {
+    GateId leftLeaf_ = inputs.find(leftLeaf)->second;
+    inputs_.push_back(Gate::Signal::always(leftLeaf_));
+    GateId rightLeaf_ = buildNet(rightLeaf, net, typeFunc, cell, inputs, typeRTLIL);
+    inputs_.push_back(Gate::Signal::always(rightLeaf_));
+    return net.addGate(func, inputs_);
+  }
+  if (!flag1 && flag2) {
+    GateId leftLeaf_ = buildNet(leftLeaf, net, typeFunc, cell, inputs, typeRTLIL);
+    inputs_.push_back(Gate::Signal::always(leftLeaf_));
+    GateId rightLeaf_ = inputs.find(rightLeaf)->second;
+    inputs_.push_back(Gate::Signal::always(rightLeaf_));
+    return net.addGate(func, inputs_);
+  }
+  if (!flag1 && !flag2) {
+    if (leftLeaf != rightLeaf) {
+      GateId leftLeaf_ = buildNet(leftLeaf, net, typeFunc, cell, inputs, typeRTLIL);
+      inputs_.push_back(Gate::Signal::always(leftLeaf_));
+      GateId rightLeaf_ = buildNet(rightLeaf, net, typeFunc, cell, inputs, typeRTLIL);
+      inputs_.push_back(Gate::Signal::always(rightLeaf_));
+    } else {
+      size_t singleLeaf = leftLeaf;
+      GateId singleLeaf_ = buildNet(singleLeaf, net, typeFunc, cell, inputs, typeRTLIL);
+      inputs_.push_back(Gate::Signal::always(singleLeaf_));
+    }
+    return net.addGate(func, inputs_);
+  }
```

Comment:
Create a dedicated procedure with an `if-else-if` control flow to avoid the warning.